ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go

```diff
@@ -4,173 +4,261 @@ import (
+		})
+	}
 }
 
-func TestRayClusterAutoscalerWithCustomResource(t *testing.T) {
-	test := With(t)
-	g := gomega.NewWithT(t)
+func TestRayClusterAutoscalerWithDesiredState(t *testing.T) {
+	for name, tc := range tests {
 
-	// Create a namespace
-	namespace := test.NewTestNamespace()
-	test.StreamKubeRayOperatorLogs()
+		test := With(t)
+		g := gomega.NewWithT(t)
 
-	// Scripts for creating and terminating detached actors to trigger autoscaling
-	scriptsAC := newConfigMap(namespace.Name, files(test, "create_detached_actor.py", "terminate_detached_actor.py"))
-	scripts, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), scriptsAC, TestApplyOptions)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
-	test.T().Logf("Created ConfigMap %s/%s successfully", scripts.Namespace, scripts.Name)
+		const maxReplica = 3
+		// Set the scale down window to a large enough value, so scale down could be disabled to avoid test flakiness.
+		const scaleDownWaitSec = 3600
 
-	test.T().Run("Create a RayCluster with autoscaling enabled", func(_ *testing.T) {
-		groupName := "custom-resource-group"
+		// Create a namespace
+		namespace := test.NewTestNamespace()
+		test.StreamKubeRayOperatorLogs()
 
+		// Scripts for creating and terminating detached actors to trigger autoscaling
+		scriptsAC := newConfigMap(namespace.Name, files(test, "create_concurrent_tasks.py"))
+		scripts, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), scriptsAC, TestApplyOptions)
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+		test.T().Logf("Created ConfigMap %s/%s successfully", scripts.Namespace, scripts.Name)
+
+		groupName := "custom-resource-group"
 		rayClusterSpecAC := rayv1ac.RayClusterSpec().
 			WithEnableInTreeAutoscaling(true).
 			WithRayVersion(GetRayVersion()).
 			WithHeadGroupSpec(rayv1ac.HeadGroupSpec().
 				WithRayStartParams(map[string]string{"num-cpus": "0"}).
-				WithTemplate(headPodTemplateApplyConfiguration())).
+				WithTemplate(tc.HeadPodTemplateGetter())).
 			WithWorkerGroupSpecs(rayv1ac.WorkerGroupSpec().
 				WithReplicas(0).
 				WithMinReplicas(0).
-				WithMaxReplicas(3).
+				WithMaxReplicas(maxReplica).
 				WithGroupName(groupName).
 				WithRayStartParams(map[string]string{"num-cpus": "1", "resources": `'{"CustomResource": 1}'`}).
-				WithTemplate(workerPodTemplateApplyConfiguration()))
+				WithTemplate(tc.WorkerPodTemplateGetter())).
+			WithAutoscalerOptions(rayv1ac.AutoscalerOptions().
+				WithIdleTimeoutSeconds(scaleDownWaitSec))
 		rayClusterAC := rayv1ac.RayCluster("ray-cluster", namespace.Name).
 			WithSpec(apply(rayClusterSpecAC, mountConfigMap[rayv1ac.RayClusterSpecApplyConfiguration](scripts, "/home/ray/test_scripts")))
 
 		rayCluster, err := test.Client().Ray().RayV1().RayClusters(namespace.Name).Apply(test.Ctx(), rayClusterAC, TestApplyOptions)
 		g.Expect(err).NotTo(gomega.HaveOccurred())
+		test.T().Logf("%s", name)
```

Comment:
You forgot to add `test.T().Run` for this test. Also, this line can be removed.

Reply:
Good catch, thank you for saving my Friday brain.